src/librbd/io/ImageDispatcher.cc

```diff
@@ -110,6 +111,55 @@ struct ImageDispatcher<I>::SendVisitor : public boost::static_visitor<bool> {
   }
 };
 
+template <typename I>
+struct ImageDispatcher<I>::ClipRequestVisitor
```

Comment:
Nit: since it's doing more than just clipping, perhaps this should be the `PreprocessVisitor` or something similar?

Reply:
Agree with you, PreprocessVisitor is better.